compiler/rustc_index/src/interval/tests.rs

```diff
@@ -0,0 +1,199 @@
+use super::*;
```

Comment:
SparseIntervalMatrix doesn't really need tests IMO -- at least currently it's a pretty thin wrapper around IntervalSet so there's not much surface area to test.

IntervalSet::union I think *currently* doesn't really merit testing -- it's a tiny wrapper around inserting all the ranges from a different set -- so I'm inclined to leave it as-is for now. If it becomes hot in some future perf assessment we may wish to optimize it by interleaving range iteration from the current and the other set (similar to a merge sort's merge operation), but that seems likely to be an obvious "we should add tests" moment), and depending on implementation would likely expose easier knobs to test -- right now it would end up pretty opaque I suspect.